server/src/test/java/org/elasticsearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -493,6 +495,84 @@ public void run() {
         assertThat(snapshotIds, either(hasSize(1)).or(hasSize(0)));
     }
 
+    public void testSuccessfulSnapshotWithConcurrentDynamicMappingUpdates() {
+        setupTestCluster(randomFrom(1, 3, 5), randomIntBetween(2, 10));
+
+        String repoName = "repo";
+        String snapshotName = "snapshot";
+        final String index = "test";
+
+        final int shards = randomIntBetween(1, 10);
+        final int documents = randomIntBetween(2, 100);
+        TestClusterNode masterNode =
+            testClusterNodes.currentMaster(testClusterNodes.nodes.values().iterator().next().clusterService.state());
+
+        final StepListener<CreateSnapshotResponse> createSnapshotResponseStepListener = new StepListener<>();
+
+        continueOrDie(createRepoAndIndex(masterNode, repoName, index, shards), createIndexResponse -> {
+            final AtomicBoolean initiatedSnapshot = new AtomicBoolean(false);
+            for (int i = 0; i < documents; ++i) {
+                masterNode.client.bulk(
+                    new BulkRequest().add(new IndexRequest(index).source(Map.of("foo" + i, "bar", "num", i)))
```

Comment:
Let's get rid of "num" field here. It seems that it's not necessary and it will make the assertion below easier (no +1). I also think that this piece of code requires some comment.

Reply:
Fair point, dropped it and added a comment